network/p2p/gossip/test_gossip.go

```diff
@@ -10,26 +10,29 @@ import (
 )
 
 var (
-	_ Gossipable   = (*testTx)(nil)
-	_ Set[*testTx] = (*testSet)(nil)
+	_ Gossipable          = (*testTx)(nil)
+	_ Set[*testTx]        = (*testSet)(nil)
+	_ Marshaller[*testTx] = (*testMarshaller)(nil)
 )
 
 type testTx struct {
 	id ids.ID
 }
 
-func (t *testTx) GetID() ids.ID {
+func (t *testTx) GetGossipID() ids.ID {
 	return t.id
 }
 
-func (t *testTx) Marshal() ([]byte, error) {
-	return t.id[:], nil
+type testMarshaller struct{}
+
+func (testMarshaller) MarshalGossip(tx *testTx) ([]byte, error) {
+	return tx.id[:], nil
 }
 
-func (t *testTx) Unmarshal(bytes []byte) error {
-	t.id = ids.ID{}
-	copy(t.id[:], bytes)
-	return nil
+func (testMarshaller) UnmarshalGossip(bytes []byte) (*testTx, error) {
+	return &testTx{
+		id: ids.ID(bytes),
+	}, nil
```

Comment:
This is supper nitty - but I think we should be using `ids.ToID` here. `ids.ID` will panic if `bytes` isn't the correct length.

Reply:
Hmm I don't think that's nitty ^_^